src/Catty/FormulaEditor/ViewController/FormulaEditorSectionViewController/FormulaEditorSectionViewController.swift

```diff
@@ -0,0 +1,226 @@
+
+        self.items = formulaManager.formulaEditorItemsForLogicSection(spriteObject: spriteObject)
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? LogicSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [LogicSubsection.logical.title, LogicSubsection.comparison.title]
+
+    }
+
+    private func initObjectItems() {
+        self.items.removeAll()
+
+        self.items = formulaManager.formulaEditorItemsForObjectSection(spriteObject: spriteObject)
+
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? ObjectSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [ObjectSubsection.general.title, ObjectSubsection.motion.title]
+
+    }
+
+    private func initSensorItems() {
+        self.items.removeAll()
+
+        self.items = formulaManager.formulaEditorItemsForDeviceSection(spriteObject: spriteObject)
+
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? SensorSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [SensorSubsection.device.title,
+                                 SensorSubsection.touch.title,
+                                 SensorSubsection.visual.title,
+                                 SensorSubsection.dateAndTime.title,
+                                 SensorSubsection.arduino.title,
+                                 SensorSubsection.phiro.title]
+    }
+
+    private func groupSubsectionWiseAndGetSize<SubsectionType: FormulaEditorSubsection & Hashable & CaseIterable>(_ subsectionType: SubsectionType?, items: inout [FormulaEditorItem]) -> [Int] {
+
+        guard let _ = subsectionType else {
+            return [Int]()
+        }
+
+        var sizes = [Int]()
+        let dict = Dictionary(grouping: items, by: { $0.sections[0].subsection() as! SubsectionType })
+        items.removeAll()
+
+        for subsection in SubsectionType.allCases {
+            if let groupedItems = dict[subsection] {
+                items.append(contentsOf: groupedItems)
+                sizes.append(groupedItems.count)
+            }
+        }
+
+        return sizes
+    }
+
+    func formulaEditorItemSelected(item: FormulaEditorItem) {
+
+        if let sensor = item.sensor {
+            self.formulaEditorVC.handleInput(for: sensor)
+        } else if let function = item.function {
+            self.formulaEditorVC.handleInput(for: function)
+        } else if let op = item.op {
+            self.formulaEditorVC.handleInput(for: op)
+        }
+
+    }
+
+    func presentUnexpectedErrorAlert() {
+        let alert = UIAlertController(title: "Some unexpected error occured", message: nil, preferredStyle: .alert)
```

Comment:
Please replace hardcoded translation with `kLocalizedUnexpectedErrorTitle` and `kLocalizedUnexpectedErrorMessage`.